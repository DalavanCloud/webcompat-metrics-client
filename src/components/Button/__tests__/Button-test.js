/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import React from "react";
import renderer from "react-test-renderer";
import sinon from "sinon";
import { shallow } from "enzyme";

import Button from "..";

it("renders Button default correctly", () => {
  const tree = renderer.create(<Button>{"I'm a Button"}</Button>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders Button: simulates handleClick events", () => {
  const onClick = sinon.spy();
  const wrapper = shallow(<Button onClick={onClick}>{"I'm a Button"}</Button>);
  wrapper.find("button").simulate("click");
  expect(onClick.calledOnce).toEqual(true);
});

it("renders Button: simulates handleBlur events", () => {
  const onBlur = sinon.spy();
  const wrapper = shallow(<Button onBlur={onBlur}>{"I'm a Button"}</Button>);
  wrapper.find("button").simulate("blur");
  expect(onBlur.calledOnce).toEqual(true);
});
