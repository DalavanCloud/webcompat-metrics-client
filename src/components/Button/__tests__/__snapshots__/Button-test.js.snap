// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Button default correctly 1`] = `
<button
  className="component"
  onBlur={[Function]}
  onClick={[Function]}
  style={Object {}}
  type="button"
>
  I'm a Button
</button>
`;
